prim.c

```diff
@@ -111,25 +111,35 @@ obj_t *primitive_add(env, args)
 obj_t *primitive_subtract(env, args)
      obj_t **env, *args;
 {
+  obj_t *node, *node_val;
   int sum = 0;
   float sumf = 0;
-  obj_t *node, *node_val;
 
-  for (node = args; node != nil; node = CDR(node)) {
+  node = args;
+  if (nil == node)
+    return alloc_int(0);
+
+  node_val = eval(CAR(node), env);
+  if (TINT == node_val->type)
+    sum += node_val->value.i;
```

Comment:
s/+=/=/ ?

Reply:
hmm

Reply:
It evaluates to
```
0 += x
```
*shrug*

If it's too many wasted cycles, I'm happy to see it changed.